lib/modules/shipments_api_retriever.rb

```diff
@@ -1,14 +1,22 @@
 module  ShipmentsApiRetriever
-
-  def self.api_call(compliance_type=nil)
-    response = HTTParty.get(ENV['SPECIES_API_URL'],
-                             headers: header,
-                             query: {compliance_type: compliance_type})
-    JSON.parse response.body
+  def self.api_call(compliance_type = nil)
+    data = []
+    page = 1
+    loop do
```

Comment:
I think it's best to handle the pagination using parameters that will be passed in to this method.
So, for example, from the `CsvDownloader` module you could loop through the number of pages required and pass in the relevant page each time to `api_call`